comanage_utils.py

```diff
@@ -81,22 +81,30 @@ def call_api2(method, target, endpoint, authstr, **kw):
 
 def call_api3(method, target, data, endpoint, authstr, **kw):
     req = mkrequest(method, target, data, endpoint, authstr, **kw)
-    trying = True
-    currentTimeout = MIN_TIMEOUT
-    while trying:
+    retries = 0
+    current_timeout = TIMEOUT_BASE
+    total_timeout = 0
+    payload = None
+    while retries <= MAX_RETRIES:
         try:
-            resp = urllib.request.urlopen(req, timeout=currentTimeout)
+            resp = urllib.request.urlopen(req, timeout=current_timeout)
             payload = resp.read()
-            trying = False
-        except urllib.error.URLError as exception:
-            if currentTimeout < MAX_TIMEOUT:
-                currentTimeout *= TIMEOUTMULTIPLE
-            else:
-                sys.exit(
-                    f"Exception raised after maximum number of retries and/or timeout {MAX_TIMEOUT} seconds reached. "
-                    + f"Exception reason: {exception.reason}.\n Request: {req.full_url}"
+            break
+        # exception catching, mainly for request timeouts and "Service Temporarily Unavailable" (Rate limiting).
+        except urllib.error.HTTPError as exception:
```

Comment:
I thought you said that we still wanted `urllib.error.URLError`? You can always just catch both